R/tm_g_distribution.R

```diff
@@ -29,11 +29,26 @@
 #'
 #' @section Decorating `tm_outliers`:
 #'
-#' This module generates the following objects, which can be modified in place using decorators:
-#' - `plot` (`ggplot2`)
+#' This module generates the following objects, which can be modified in place using decorators::
+#' - `plot` (`ggplot2`)`
 #' - `test_table` (`data.frame`)
 #' - `summary_table` (`data.frame`)
 #'
+#' The `decorators` can be specific for the `histogram` and `decorators` outputs
+#' within the module, which can be specified by using a named list.
+#' See code snippet below:
+#'
+#' ```
+#' tm_g_distribution(
+#'    ..., # arguments for module
+#'    decorators = list(
+#'      default = list(teal_transform_module(...)), # applied to outputs of both `histogram` and `qq`
+#'      histogram = list(teal_transform_module(...)), # applied to outputs of `histogram`
+#'      qq = list(teal_transform_module(...)) # applied to outputs of `qq`
+#'    )
+#' )
+#' ```
+#'
```

Comment:
@m7pr @llrs-roche I'd appreciate an opinion on how to best describe the named parameters for the decorators.

Reply:
@averissimo I think we should include those in the vignette.

I think you also wanted to write

```r
#' The `decorators` can be specific for the `histogram` and `qq` outputs
```

instead of 

```r
#' The `decorators` can be specific for the `histogram` and `decorators` outputs
```

but I think something is off.

I think this should be like below

```r
#' tm_g_distribution(
#'    ..., # arguments for module
#'    decorators = list(
#'      default = list(teal_transform_module(...)), # applied to all outputs
#'      plot = list(teal_transform_module(...)), # applied only to plot output (histogram plot)
#'      test_table= list(teal_transform_module(...)), # applied only to test_table
#'      summary_table= list(teal_transform_module(...)), # applied only to summary_table
#'    )
#' )
```

Let's don't introduce another naming convention, where we also bind the internal objects names (like plot) with output UI names (like histogram).

Reply:
I think the description is correct enough for now. Later, with more user feedback we could describe and rule out some cases/patterns. But I like the simplified approach of just the name of the object to be applied on the decorators list. How its intended use is: Will users want multiple decorators for a given object or just one would be enough? This could lead to a slight simplification:

```
#' tm_g_distribution(
#'    ..., # arguments for module
#'    decorators = list(
#'      default = teal_transform_module(...), # applied to all outputs
#'      plot = teal_transform_module(...), # applied only to plot output (histogram plot)
#'      test_table= teal_transform_module(...), # applied only to test_table
#'      summary_table= teal_transform_module(...), # applied only to summary_table
#'    )
#' )
```

Other things that might be worth documenting (I haven't checked what happens) if we have a default decorator and also a specific decorator for an object, which one will be applied first? If there are multiple which is the order or if there are 3 defaults decorators and 2 specific decorators, how are they combined?